ccmpred/scripts/run_ccmgen.py

```diff
@@ -212,9 +212,22 @@ def main():
                 neff = ccm.neff_entropy
             else:
                 neff = opt.neff
-
-            msa_sampled, neff = ccmpred.sampling.sample_to_neff_increasingly(
-                tree, nseq, neff, ncol, x, opt.seq0_mrf)
+            
+            if opt.seq0_file:
+                
+                seq0 = ccmpred.io.alignment.read_msa(opt.seq0_file, opt.aln_format)
```

Comment:
the logic of reading and validating the root sequence seems duplicated here. I would extract it to a function 
```python
def read_root_sequence(seq0_file, aln_format, print_sequence=True):
    seq0 = ccmpred.io.alignment.read_msa(seq0_file, aln_format)
    seq_N, seq_L = seq0.shape
    
    if seq_L != n_col:
       [...]
 
    # we probably also have to ensure that seq_N == 1?

     if print_sequence:
          print("Ancestor sequence:\n{0}".format("".join([ccmpred.io.alignment.AMINO_ACIDS[c] for c in seq0[0]])))


    return seq0

    
```

Reply:
I did not where to put this function, so I put it inside the main function. Maybe it is not the cleanest thing to do ?